### PR TITLE
Detecting Data Exfiltration via DevTunnels/Forwarded Ports

### PR DESCRIPTION
**Pull request by deFr0ggy**
### Summary of the Pull Request
This PR adds the following:

- 2 new rules for the `network_connection` event log that focus on initiated communication by a process to either Devtunnels or Vscode tunnel domains.
- It also split the logic of both `dns_query` and `network_connection` rules related to VsCode/Devtunnels (by @nasbench)

### Changelog

new: DNS Query To Devtunnels Domain - Split rule based on `b3e6418f-7c7a-4fad-993a-93b65027a9f1`
new: Network Connection Initiated To DevTunnels Domain
new: Network Connection Initiated To Visual Studio Code Tunnels Domain
update: DNS Query To Visual Studio Code Tunnels Domain - Update the rule to only focus on DNS requests from Vscode tunnels and move the logic of Devtunnels to another rule. To ease FP management for users that leverage one but not the other.

### Example Log Event

NA
### Fixed Issues
NA

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

**Review by nasbench — APPROVED** [review]

**Comment by nasbench** [comment]
Thanks for this contribution @deFr0ggy and welcome to the Sigma community. Just a quick note on the changes i've made so that it's clear.

There was already a rule focusing on DNS requests for both Devtunnels and Vscode tunnel domains. I decided to split your rule and the previous one into separate rules just because of FP management and expectations.

This would leave it to the user to choose a rule depending on what he uses and give them the ability to increase the level and management FP filters if for exmaple Vscode tunnels is used but not Devtunnels (and vice versa).

As for your original rule since this is your first contribution. I just wanted to highlight that when you use the `endswith` modifier it adds a `*` (wildcard) at the start. So you could've ignored the rest of the domains and only focus on unique endings.

So this example

```yml
            - '.devtunnels.ms'
            - '.euw.devtunnels.ms'
            - '.usw2.devtunnels.ms'
            - '.tunnels.api.visualstudio.com'
            - '.rel.tunnels.api.visualstudio.com'
```

Becomes like this
```yml
            - '.devtunnels.ms'
            - '.tunnels.api.visualstudio.com'
```

Thanks once again.

File: rules/windows/dns_query/dns_query_win_devtunnels_communication.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+related:
+    - id: 9501f8e6-8e3d-48fc-a8a6-1089dd5d7ef4 # Net Connection DevTunnels
+      type: similar
+    - id: 4b657234-038e-4ad5-997c-4be42340bce4 # Net Connection VsCode
+      type: similar
+    - id: b3e6418f-7c7a-4fad-993a-93b65027a9f1 # DNS VsCode
+      type: similar
+status: experimental
+description: |
+    Detects DNS query requests to Devtunnels domains. Attackers can be abuse such feature to establish a reverse shell or persiste on a machine.
```
Comment (phantinuss) [inline rules/windows/dns_query/dns_query_win_devtunnels_communication.yml:12]:
```suggestion
    Detects DNS query requests to Devtunnels domains. Attackers can abuse that feature to establish a reverse shell or persistence on a machine.
```

File: rules/windows/dns_query/dns_query_win_vscode_tunnel_communication.yml [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
+    - id: 9501f8e6-8e3d-48fc-a8a6-1089dd5d7ef4 # Net Connection DevTunnels
+      type: similar
+    - id: 4b657234-038e-4ad5-997c-4be42340bce4 # Net Connection VsCode
+      type: similar
+    - id: 1cb0c6ce-3d00-44fc-ab9c-6d6d577bf20b # DNS DevTunnels
+      type: similar
 status: experimental
 description: |
-    Detects DNS query to Devtunnels and Visual Studio Code tunnel domains. Attackers can be abuse these features to establish a reverse shell.
+    Detects DNS query requests to Visual Studio Code tunnel domains. Attackers can be abuse such feature to establish a reverse shell or persiste on a machine.
```
Comment (phantinuss) [inline rules/windows/dns_query/dns_query_win_vscode_tunnel_communication.yml:12]:
```suggestion
    Detects DNS query requests to Visual Studio Code tunnel domains. Attackers can abuse that feature to establish a reverse shell or persistence on a machine.
```

File: rules/windows/network_connection/net_connection_win_devtunnel_connection.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+related:
+    - id: 4b657234-038e-4ad5-997c-4be42340bce4 # Net Connection VsCode
+      type: similar
+    - id: b3e6418f-7c7a-4fad-993a-93b65027a9f1 # DNS VsCode
+      type: similar
+    - id: 1cb0c6ce-3d00-44fc-ab9c-6d6d577bf20b # DNS DevTunnels
+      type: similar
+status: experimental
+description: |
+    Detects network connections to Devtunnels domains initated by a process on a system. Attackers can be abuse such feature to establish a reverse shell or persiste on a machine.
```
Comment (phantinuss) [inline rules/windows/network_connection/net_connection_win_devtunnel_connection.yml:12]:
```suggestion
    Detects network connections to Devtunnels domains initiated by a process on a system. Attackers can abuse that feature to establish a reverse shell or persistence on a machine.
```

File: rules/windows/network_connection/net_connection_win_vscode_tunnel_connection.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+related:
+    - id: 9501f8e6-8e3d-48fc-a8a6-1089dd5d7ef4 # Net Connection DevTunnels
+      type: similar
+    - id: b3e6418f-7c7a-4fad-993a-93b65027a9f1 # DNS VsCode
+      type: similar
+    - id: 1cb0c6ce-3d00-44fc-ab9c-6d6d577bf20b # DNS DevTunnels
+      type: similar
+status: experimental
+description: |
+    Detects network connections to Visual Studio Code tunnel domains initated by a process on a system. Attackers can be abuse such feature to establish a reverse shell or persiste on a machine.
```
Comment (phantinuss) [inline rules/windows/network_connection/net_connection_win_vscode_tunnel_connection.yml:12]:
```suggestion
    Detects network connections to Visual Studio Code tunnel domains initiated by a process on a system. Attackers can abuse that feature to establish a reverse shell or persistence on a machine.
```

**Review by phantinuss — APPROVED** [review]

**Pull request merged**